### ISSUE TITLE
Small bug in autodiff.py (I think)

### ISSUE THREAD

**Issue by dlichy** [mitsuba-renderer/mitsuba2#51]
First I want to congratulate you on the release of Mitsuba 2. I'm a researcher in 3D reconstruction and I think it will be an invaluable tool in the field.

I believe there is a small bug in "mitsuba2/src/python/python/autodiff.py" on line 402. You have the chunk of code: 
```
for k, v in args.items():
                        if k == 'spp':
                            spp = v
                        elif k == 'sensor_index':
                            sensor_index = v
                        elif k == 'unbiased':
                            unbiased = v
                        elif k == 'malloc_trim':
                            malloc_trim = v
                        elif params is not None:
                            params[k] = type(params[k])(v)
                            ctx.inputs.append(None)
                            ctx.inputs.append(params[k] if v.requires_grad
                                              else None)
                        ctx.inputs.append(None) <-- these should only be executed if the last elif is not executed.
                        ctx.inputs.append(None)
```
I think this needs to be changed to something like:
```
 for k, v in args.items():
                        if k == 'spp':
                            spp = v
                            ctx.inputs.append(None)
                            ctx.inputs.append(None)
                        elif k == 'sensor_index':
                            sensor_index = v
                            ctx.inputs.append(None)
                            ctx.inputs.append(None)
                        elif k == 'unbiased':
                            unbiased = v
                            ctx.inputs.append(None)
                            ctx.inputs.append(None)
                        elif k == 'malloc_trim':
                            malloc_trim = v
                            ctx.inputs.append(None)
                            ctx.inputs.append(None)
                        elif params is not None:
                            params[k] = type(params[k])(v)
                            ctx.inputs.append(None)
                            ctx.inputs.append(params[k] if v.requires_grad
                                              else None)
```
Otherwise you get an error "function RenderBackward returned an incorrect number of gradients" when you keep more than one differentiable parameter in your scene.

Sorry the white space didn't format well. Hopefully you see what I mean. I marked it with a <--

**Comment by Speierers** [comment]
This looks good to me. @wjakob could you confirm? I will be happy to push those changes.

**Comment by wjakob** [comment]
How about just adding a `continue` after the last `else` stmt? I think that's how it was at some point to minimize replicating many repetitions of ``ctx.inputs.append(None)``.

**Comment by Speierers** [comment]
Fixed here: 008cb4d
Thanks @dlichy !
